pkg/arvo/tests/sys/zuse/crypto/ed25519.hoon

```diff
@@ -0,0 +1,156 @@
+::  tests for the ed25519 signature algorithm
+::
+/+  *test
+=,  ed:crypto
+::
+|%
+::
+::  Test vectors from Section 7.1 of RFC 8032:
+::  https://tools.ietf.org/html/rfc8032#section-7.1
+::
++$  vector  [seed=@ux pk=@ux msg=@ux sig=@ux]
+::
+++  test-vectors
+  ^-  tang
+  |^  ;:  weld
+        %+  category  "puck"
+        (zing (turn vectors check-puck))
+        %+  category  "sign"
+        (zing (turn vectors check-sign))
+        %+  category  "veri"
+        (zing (turn vectors check-veri))
+      ==
+  ::
+  ++  check-puck
+    |=  vector
+    %+  expect-eq
+      !>  pk
+      !>  `@ux`(puck seed)
+  ::
+  ++  check-sign
+    |=  vector
+    %+  expect-eq
+      !>  sig
+      !>  `@ux`(sign msg seed)
+  ::
+  ++  check-veri
+    |=  vector
+    %-  expect
+      !>  (veri sig msg pk)
+  ::
+  ++  vectors
+    (turn vectors-raw swp-vec)
```

Comment:
Not strictly necessary, but a neat trick: you can turn this into a constant via compile-time evaluation with `^~`:

Compare:

```
> =<  -  |%  ++  foo  (turn bar |=(@ +(+<)))  ++  bar  (gulf 0 4)  --
[[8 [9 3.075.070 0 1.023] 9 2 10 [6 [7 [0 3] 1 0] 7 [0 3] 1 4] 0 2] 8 [9 96.126 0 1.023] 9 2 10 [6 [7 [0 3] 9 4 0 1] 7 [0 3] 8 [1 0] [1 4 0 6] 0 1] 0 2]
```

with

```
> =<  -  |%  ++  foo  ^~  (turn bar |=(@ +(+<)))  ++  bar  ^~  (gulf 0 4)  --
[[1 0 1 2 3 4 0] 1 1 2 3 4 5 0] 
```

Reply:
neat! I think it's appropriate here.